src/lib.rs

```diff
@@ -2884,7 +2849,10 @@ pub unsafe trait AsBytes {
     /// assert_eq!(excessive_bytes, [0u8; 128]);
     /// ```
     #[inline]
-    fn write_to(&self, bytes: &mut [u8]) -> Option<()> {
+    fn write_to(&self, bytes: &mut [u8]) -> Option<()>
+    where
+        Self: NoCell,
```

Comment:
Ideally we wouldn't need this bound, but I'm not sure how to implement this method soundly without it. Thankfully this isn't a regression (in 0.7.X, `AsBytes` forbids `UnsafeCell`s full stop). However, at a minimum, we may want to file an issue to track eventually relaxing this bound (and the `NoCell` bound on other `write_to_xxx` methods).

Reply:
Filed https://github.com/google/zerocopy/issues/688 to track.